buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchDistribution.java

```diff
@@ -217,22 +238,36 @@ public Configuration getExtracted() {
 
     @Override
     public TaskDependency getBuildDependencies() {
```

Comment:
I don't think either of these methods should be public. If I need to wire up dependencies I should depend on either the Distribution itself or `getExtracted` depending on what I'm using and then just rely on implicit dependencies.

Reply:
We have this public as it's required as Distribution does (and has before) implemented `Buildable`

Reply:
getBuildDependencies is required as `ElasticsearchDistribution` is implementing `Buildable`, so we cannot remove this without major rework. While working on this I noticed that `ElasticsearchDistribution` does a little bit too much at the moment and in one point we might want to split this up a bit one day